src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,124 @@
 package core.basesyntax;
 
 public class ArrayList<T> implements List<T> {
+    private static final double GROWTH_INDEX = 1.5;
+    private static final int INITIAL_CAPACITY = 10;
+    private Object[] elements;
+    private int numberOfElements;
+
+    public ArrayList() {
+        elements = new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        checkCapacity(numberOfElements + 1, elements.length);
+        elements[numberOfElements] = value;
+        numberOfElements++;
     }
 
     @Override
     public void add(T value, int index) {
+        checkCapacity(numberOfElements + 1, elements.length);
+        if (index == numberOfElements) {
+            elements[index] = value;
+            numberOfElements++;
+            return;
+        } else if (index != 0) {
+            rangeIndexCheck(index);
+        }
 
+        for (int i = numberOfElements; i > index; i--) {
+            elements[i] = elements[i - 1];
+        }
+        elements[index] = value;
+        numberOfElements++;
     }
 
     @Override
     public void addAll(List<T> list) {
-
+        checkCapacity(numberOfElements + list.size(), list.size());
+        for (int i = 0; i < list.size(); i++) {
+            elements[numberOfElements++] = list.get(i);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        rangeIndexCheck(index);
+        return (T) elements[index];
     }
 
     @Override
     public void set(T value, int index) {
-
+        rangeIndexCheck(index);
+        elements[index] = value;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        rangeIndexCheck(index);
+        final T removedElement = (T)elements[index];
+
+        for (int i = index + 1; i < numberOfElements; i++) {
+            elements[i - 1] = elements[i];
+        }
+        elements[numberOfElements - 1] = null;
+        numberOfElements--;
+        return removedElement;
     }
 
     @Override
     public T remove(T element) {
-        return null;
+        int index = findIndex(element);
+        if (index != -1) {
+            remove(index);
+        } else {
+            throw new java.util.NoSuchElementException("No such element found");
```

Comment:
This error message can be constant - ERROR_MESSAGE.
And also i think checkCapasity and calculateCapasity can be 1 method because you will increase capasity anyway if actualcapasity if lower.